actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -427,49 +427,38 @@ module FormHelper
       def form_for(record, options = {}, &block)
         raise ArgumentError, "Missing block" unless block_given?
         html_options = options[:html] ||= {}
+        html_options.merge!(options.extract!(:data, :method, :enforce_utf8, :authenticity_token))
```

Comment:
We should not be mutating the input with this `merge!`

Reply:
@rafaelfranca I've pushed up some changes to incorporate your code review.

I wasn't sure how to do this in a way that doesn't mutate the user-provided `options[:html]` hash. Do you have something in mind?